contracts/CPKFactoryFacade.sol

```diff
@@ -0,0 +1,57 @@
+pragma solidity >=0.5.0 <0.7.0;
+
+import { Enum } from "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+import { CPKFactory } from "./CPKFactory.sol";
+
+contract CPKFactoryFacade {
+    CPKFactory cpkFactory;
+    address safeVersion;
+    uint256 saltNonce;
+    address fallbackHandler;
+
+    constructor(
+        CPKFactory _cpkFactory,
+        address _safeVersion,
+        uint256 _saltNonce,
+        address _fallbackHandler
+    ) public {
+        cpkFactory = _cpkFactory;
+        safeVersion = _safeVersion;
+        saltNonce = _saltNonce;
+        fallbackHandler = _fallbackHandler;
+    }
+
+    function execTransaction(
+        address to,
+        uint256 value,
+        bytes calldata data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address payable refundReceiver,
+        bytes calldata signature
+    )
+        external
+        payable
+        returns (bool)
+    {
+        address owner;
+        // solium-disable-next-line security/no-inline-assembly
+        assembly {
+            owner := and(
+                calldataload(add(calldataload(0x124), 0x61)),
```

Comment:
missing some comments here :D

Reply:
so we have 9 * 20 + 4 => 124 (all in hex)
the 61 is the length if 1 signature ... that assumes that we have threshold 1. Wouldn't it be better to do 
`calldataload(add(calldataload(0x124), calldataload(calldataload(0x124)))),` (or similar ... so load the length of the signatures bytes and add them to get the end of the bytes.

More general comment ... this is super hacky (adding the address as an "unused" param :P )

Reply:
I made a whole utils file to try and do this without the super hacky bit and found a circular dependency, so this approach I think will have to stay as far as the facade goes. I think this also requires allowing these special signatures in the Safe relay service to pass. Anyway, the utils file is here:

<details>
```solidity
pragma solidity >=0.5.0 <0.7.0;

import { GnosisSafeProxyFactory } from "@gnosis.pm/safe-contracts/contracts/proxies/GnosisSafeProxyFactory.sol";
import { SignatureDecoder } from "@gnosis.pm/safe-contracts/contracts/common/SignatureDecoder.sol";

contract SafeUtils is SignatureDecoder {
    bytes32 public constant SAFE_TX_TYPEHASH = keccak256(
       "SafeTx(address to,uint256 value,bytes data,uint8 operation,uint256 safeTxGas,uint256 baseGas,uint256 gasPrice,address gasToken,address refundReceiver,uint256 nonce)"
    );

    bytes32 public constant DOMAIN_SEPARATOR_TYPEHASH = keccak256(
        "EIP712Domain(address verifyingContract)"
    );

    function calculateCreateProxyWithNonceAddress(
        GnosisSafeProxyFactory safeFactory,
        address impl,
        bytes memory initializer,
        uint256 saltNonce
    )
        internal
        pure
        returns (address)
    {
        return address(uint(keccak256(abi.encodePacked(
            // create2 address calculation prefix
            hex"ff",
            // deployer
            safeFactory,
            // salt
            keccak256(abi.encodePacked(keccak256(initializer), saltNonce)),
            // deployment bytecode
            abi.encodePacked(safeFactory.proxyCreationCode(), uint256(impl))
        ))));
    }
}
```
</details>